src/include/migraphx/op/random_uniform.hpp

```diff
@@ -75,7 +75,16 @@ struct random_uniform
 
         result.visit([&](auto output) {
             using type = typename decltype(output)::value_type;
-            if constexpr(std::is_integral<type>{})
+            if constexpr(
+                std::is_integral_v<type>
+#ifdef _MSC_VER
+                // According to the C++ specification, the effect is undefined if the result type
+                // for the generator is not one of short, int, long, long long, unsigned short,
+                // unsigned int, unsigned long, or unsigned long long. See
+                // https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution.
+                && !(std::is_same_v<type, unsigned char> || std::is_same_v<type, signed char>)
```

Comment:
This will produce wrong results for int8 on windows. When we are using an unsupported type then we need to use a supported one and then clamp it to the supported:

```cpp
template<class T>
using uniform_int_distribution = std::uniform_int_distribution<std::conditional_t<sizeof(T) == 1, int, T>>;
```

And then clamp it when we construct it:

```cpp
uniform_int_distribution<type> dis(0, std::numeric_limits<type>::max());
std::generate(output.begin(), output.end(), [&] { return dis(gen); });
```

Reply:
That make sense. I will try it and let you know the result.

Reply:
> Then we need to use a supported one and then clamp it to the supported: 

That would violate how a random uniform should work.  It should produce uniformly distributed numbers across the range. 

I think we can just not support it for int8 for now. ONNX only supports for real values.

Reply:
> That would violate how a random uniform should work. It should produce uniformly distributed numbers across the range.

Yea "clamping" was probably a poor choice of words.

> or otherwise we would need to scale it down to range of int8 and then quantize/round it to nearest int

We dont need to do that, we can pass the range we want to the constructor. This should produce the same results.

Originally we were relying on the default constructor to set the range based on the data type, but since we are using a different generator type then we need to pass the range to the constructor.

Reply:
I strongly recommend not trying to support anything not already supported by STL or other libraries.  There will be a mess of special and edge cases to consider, and any shortcoming will create subtle statistical biases that could turn into problems for our future customers at runtime.  Coding statistical library functions is a specialized field that we do not want to dabble in.

Reply:
A little more reading in StackOverflow indicates that the C++ committee never gave a mathematical reason for disallowing int8 types for `uniform_int_distribution`.  So it could be implemented by instantiating `uniform_int_distribution` with range arguments(0, 255) in the contructor and then assigning to the desired type.